### PR TITLE
Tableau lineage BigQuery use project Id [sc-6500]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

The Tableau API is returning the BigQuery project name as database name. This is not consistent with BQ crawler which uses project ID.

### 🤓 What?

- Add optional config to allow the user to provide the big query project ID, since it's difficult, hacky and unreliable to parse the info from the Tableau data sources API response.
- Add more logs in zip file to aid the debugging of missing lineage

### 🧪 Tested?

tested on local run, and verified the dataset ID matches the one from BQ crawler

### CODE REVIEW

File: metaphor/tableau/README.md [hunk 1]
```diff
@@ -67,6 +67,14 @@ If one of the data sources is using Snowflake dataset, please provide the Snowfl
 snowflake_account: <account_name>
 ```
 
+#### BigQuery project ID
+
+If one of the data sources is using BigQuery dataset, please provide the BigQuery project ID (NOT project name) as follows,
+
+```yaml
+bigquery_project_id: <project_id>
```
Comment (mars-lan) [inline metaphor/tableau/README.md:75]: This should be a map of project names to project IDs

Comment (alyiwang) [reply metaphor/tableau/README.md]: updated

File: metaphor/tableau/README.md [hunk 2]
```diff
@@ -67,6 +67,16 @@ If one of the data sources is using Snowflake dataset, please provide the Snowfl
 snowflake_account: <account_name>
 ```
 
+#### BigQuery project ID
+
+If one of the data sources is using BigQuery dataset, please provide the BigQuery project name to project ID map as follows,
```
Comment (mars-lan) [inline metaphor/tableau/README.md:72]: See if you can find good links for BQ project name & ID?

File: metaphor/bigquery/lineage/extractor.py [hunk 3]
```diff
@@ -171,9 +170,6 @@ def _parse_job_change_entry(self, entry: LogEntry):
             logger.info(f"Skipped table: {destination.table_name()}")
             return
 
-        if job_change.user_email in self._excluded_usernames:
```
Comment (mars-lan) [inline metaphor/bigquery/lineage/extractor.py:174]: Why removing this?

File: metaphor/tableau/config.py [hunk 4]
```diff
@@ -32,5 +32,8 @@ class TableauRunConfig(BaseConfig):
     access_token: Optional[TableauTokenAuthConfig]
     user_password: Optional[TableauPasswordAuthConfig]
 
-    # snowflake data source account
-    snowflake_account: Optional[str]
+    # Snowflake data source account
+    snowflake_account: Optional[str] = None
+
+    # BigQuery data source project name to project ID map
+    bigquery_project_name_to_id_map: Optional[Dict] = None
```
Comment (mars-lan) [inline metaphor/tableau/config.py:39]: Should probably initialize this to an empty map to avoid crashing when doing membership testing?

Comment (alyiwang) [reply metaphor/tableau/config.py:39]: good point! updated

File: metaphor/tableau/README.md (hunk 2 above)
Comment (alyiwang) [reply metaphor/tableau/README.md:72]: added

File: metaphor/bigquery/lineage/extractor.py (hunk 3 above)
Comment (alyiwang) [reply metaphor/bigquery/lineage/extractor.py:174]: In https://github.com/MetaphorData/connectors/pull/147 we discussed to remove `excluded_usernames` and may add `include_username` in future. I think some changes are left out in that PR, but a `git add -A` added them here...

File: metaphor/tableau/extractor.py [hunk 5]
```diff
@@ -106,6 +114,7 @@ async def extract(self, config: TableauRunConfig) -> List[MetadataChangeEvent]:
             # by the REST api, can NOT use id to match entities.
             resp = server.metadata.query(workbooks_graphql_query)
             resp_data = resp["data"]
+            logger.debug(json.dumps(resp_data))
```
Comment (mars-lan) [inline metaphor/tableau/extractor.py:117]: Add some context for the log as well?

File: metaphor/tableau/extractor.py [hunk 6]
```diff
@@ -57,6 +59,10 @@ async def extract(self, config: TableauRunConfig) -> List[MetadataChangeEvent]:
         logger.info("Fetching metadata from Tableau")
 
         self._snowflake_account = config.snowflake_account
+        if config.bigquery_project_name_to_id_map is not None:
```
Comment (mars-lan) [inline metaphor/tableau/extractor.py:62]: No longer needed?

File: metaphor/tableau/config.py (hunk 4 above)
Comment (mars-lan) [reply metaphor/tableau/config.py]: Actually you should do something like this: https://github.com/MetaphorData/connectors/blob/main/metaphor/bigquery/config.py#L21

File: metaphor/tableau/extractor.py (hunk 6 above)
Comment (alyiwang) [reply metaphor/tableau/extractor.py]: right, removed

File: metaphor/tableau/extractor.py [hunk 7]
```diff
@@ -79,21 +82,25 @@ async def extract(self, config: TableauRunConfig) -> List[MetadataChangeEvent]:
...
             # fetch all workbooks
-            workbooks: List[WorkbookItem] = [item for item in Pager(server.workbooks)]
+            workbooks: List[WorkbookItem] = list(Pager(server.workbooks))
             logger.info(
                 f"\nThere are {len(workbooks)} work books on site: {[workbook.name for workbook in workbooks]}"
             )
             for item in workbooks:
                 server.workbooks.populate_views(item, usage=True)
+                # log workbook item detail for debugging purpose
+                logger.debug(json.dumps(item.__dict__, default=str))
```
Comment (mars-lan) [inline metaphor/tableau/extractor.py:103]: What I meant is the log itself should include some context, e.g. `logger.debug(f'workbook item: {json.dump(...)})`

File: pyproject.toml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "metaphor-connectors"
-version = "0.10.9"
+version = "0.10.10"
```
Comment (mars-lan) [inline pyproject.toml:3]: 11

File: metaphor/tableau/config.py (hunk 4 above)
Comment (alyiwang) [reply metaphor/tableau/config.py]: Seems the same as what I had here? 😄  ```
dataclasses.field(
        default_factory=dict
    )
    ```

Comment (mars-lan) [reply metaphor/tableau/config.py]: I must have misread it or was reading an older verison.

File: metaphor/tableau/extractor.py (hunk 7 above)
Comment (alyiwang) [reply metaphor/tableau/extractor.py:102]: I prefer not to add this inline context: 1. there is some context above `There are {len(workbooks)} workbooks on site`. 2. Wrap JSON in string causes all the quotes to be eacaped, making it hard to read, and requiring an extra step to unescape the string then prettify it.

File: pyproject.toml (hunk 8 above)
Comment (alyiwang) [reply pyproject.toml]: good catch, merged main

File: metaphor/tableau/extractor.py (hunk 7 above)
Comment (mars-lan) [reply metaphor/tableau/extractor.py:102]: SGTM

File: tests/tableau/test_config.py [hunk 9]
```diff
@@ -13,7 +13,28 @@ def test_yaml_config(test_root_dir):
             token_value="bar",
         ),
         user_password=None,
-        disable_preview_image=True,
```
Comment (mars-lan) [inline tests/tableau/test_config.py:16]: add it back?

**Review by mars-lan — COMMENTED** [review]
Otherwise LGTM

File: tests/tableau/test_config.py (hunk 9 above)
Comment (alyiwang) [reply tests/tableau/test_config.py:16]: It's been merged to below, not removed

**Review by mars-lan — APPROVED** [review]

**Pull request merged**